File: app/javascript/serializers/MenuAdminSerializer.ts
export type MenuAdminSerializer = {
  id: number;
  department: string;
  start_at: Date;
  end_at: Date;
  appointment_id: number;
};

export const castToMenuAdminSerializer = (data: any): MenuAdminSerializer => {
  return {
    id: data.id,
    department: data.department,
    start_at: new Date(data.start_at),
    end_at: new Date(data.end_at),
    appointment_id: data.appointment_id,
  };
};

File: app/serializers/menu_admin_serializer.rb
# frozen_string_literal: true

class MenuAdminSerializer < ActiveModel::Serializer
  attribute :id
  attribute :department
  attribute :start_at
  attribute :end_at
  attribute :appointment_id

  def appointment_id
    return nil if object.appointment.empty?

    object.appointment.first.id
  end
end
